server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2911,9 +2918,40 @@ public UserVm updateVirtualMachine(long id, String displayName, String group, Bo
             }
         }
 
+        if (!StringUtils.isEmpty(hostName)) {
+            vm.setHostName(hostName);
+            try {
+                updateDns(vm);
+            } catch (Exception e) {
+                throw new CloudRuntimeException(String.format("Failed to update hostname of VM %s to %s", vm.getInstanceName(), vm.getHostName()));
+            }
```

Comment:
@Pearl1594 do we need to really catch everything here?

Reply:
not necessarily actually